### ISSUE TITLE
MQTT stops receiving messages, but can publish

### ISSUE THREAD

**Issue by adams-family** [node-red/node-red#4138]
### Current Behavior

I've got both NodeRed (`v2.2.2`) and MQTT running in docker containers, hosted locally.

- Everything was running fine for months but suddenly stopped receiving messages from MQTT - verified with this simple flow, this did not output anything to the debug console:

<img width="331" alt="image" src="https://user-images.githubusercontent.com/19559733/232424424-e7d1a74f-3bd6-44fc-bc03-0ea0a5ac965d.png">

- At the same time, it was able to publish messages to the same MQTT with no issues - tried it several hundred times (literally) and verified it on the MQTT console.

- Other application were communicating fine via MQTT, for example zigbee2mqtt was fine both receiving and publishing.

- Restarting the MQTT server did not help.

**All the above** lead me to the conclusion that the problem must be in NodeRED. I restarted NodeRED and everything started working fine.

I've got NodeRED logs, but theres a lot of them, I'm not sure if they are useful. Any hints on how to debug this?

### Expected Behavior

NodeRed should receive messages from MQTT.

### Steps To Reproduce

Happens randomly, can't reproduce.

### Example flow

Just print out all MQTT messages.

<img width="331" alt="image" src="https://user-images.githubusercontent.com/19559733/232424606-37c847c5-2f8b-430d-ac43-7acd9a3c45b1.png">

### Environment

- Node-RED version: 2.2.2 from Docker Hub
- Node.js version: (as on Docker Hub)
- npm version: (as on Docker Hub)
- Platform/OS: docker image from Docker Hub
- Browser: Chrome on macOS

**Comment by Steve-Mcl** [comment]
> * At the same time, it was able to publish messages to the same MQTT with no issues - tried it several hundred times (literally) and verified it on the MQTT console.

Did you try editing the config node (i.e. change its name or add 1 to the keep alive) then do a "Node Deploy"?  That would  destroy and re-create the MQTT client and its connection - and would add weight to your theory.  

> All the above lead me to the conclusion that the problem must be in NodeRED. I restarted NodeRED and everything started working fine.

How did you restart Node-RED? Did you stop/start the container? did you do a full deploy? did you do a "restart flows"?

---

Alternatively: 
Is it possible you could upgrade to Node-RED current (3.0.2) and see if your issues persist?

**Comment by adams-family** [comment]
> Did you try editing the config node (i.e. change its name or add 1 to the keep alive) then do a "Node Deploy"? That would destroy and re-create the MQTT client and its connection - and would add weight to your theory.

No, unfortunately :( I opened the MQTT node but I didn't know how to force a reconnect. Will do next time.

> How did you restart Node-RED? Did you stop/start the container? did you do a full deploy? did you do a "restart flows"?

I did:

```docker compose restart nodered```

> Is it possible you could upgrade to Node-RED current (3.0.2) and see if your issues persist?

I upgraded to 2.2.3-12 but I need to wait with upgrading to v3 for the weekend, as I have dozens of flows, a lot of integrations and will need to retest everything.

It's hard to tell if the problem persists, as this happened after a few weeks of uptime :( It will take a long time to conclude that the problem is gone. However, if we can debug what happened today, I'm happy to dig deeper.

**Comment by Steve-Mcl** [comment]
@thalesmaoa thanks for chipping in.

> I confirm that, from time to time, I have the same problem

Can you please confirm Node-RED version is the same (2.2.2) as the OP (or are you on a different version? - please state)

**Comment by thalesmaoa** [comment]
I took some time to test and I notice that, in my setup, this problem is associated with high usage of processor and lost of connection.

When I have the same problem as @adams-family, I deploy and it start working again. Also, a browser refresh works also. So, I decided to flood the network with messages. 

```
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import asyncio
import asyncio_mqtt as aiomqtt

async def main():
    async with aiomqtt.Client(hostname="10.254.0.253", username="admin",password="Laj1nh0!") as client:
        count = 0
        while True:
            count += .1
            asyncio.create_task( client.publish("driver/Lab01", payload=count) )

asyncio.run(main())
```

After some time it stops feeding. 
I'm using the last version of mosquitto and nodered in different docker containers. Also, my hardware is Raspberry Pi 3.

In my case, I can't confirm that this is a nodered problem.

**Comment by Steve-Mcl** [comment]
@thalesmaoa 

> Also, a browser refresh works also

That suggests the issue is somewhere between the websocket ~ debug panel (the websocket node-red feeds debug messages to the browser)

In short, the subscription may very well still be OK and it is simply the front end visualisation that is problematic.
This can easily be verified by using a debug that outputs to console (i.e. serverside).

**Comment by luixal** [comment]
Hi,

Just hit this issue. I can confirm this is still happening in version 3.1.1. In my case, no message was received at all, didn't matter I restarted the server, refreshed the browser or whatever, zero messages logged in the browser.

I can also confirm it's a UI issue, mqtt messages are received and logs can be seen serverside.

Really annoying though as sometimes you only have access via browser and not ssh.

**Comment by knolleary** [comment]
Closing as part of a tidy-up of the issue tracker — this has been inactive for a long time and much has changed in Node-RED since. If it's still relevant on a current version, please open a fresh issue with up-to-date details and we'll take a look. Thanks for raising it.
